From a76488a84db452865e114336b3fde6257e327715 Mon Sep 17 00:00:00 2001
From: Oliver Smith <oliver@8.c.9.b.0.7.4.0.1.0.0.2.ip6.arpa>
Date: Fri, 11 Oct 2013 16:41:12 +0200
Subject: [PATCH 1/2] netfilter: ipset: Fix configure failure when
 --with-kmod=no

When configuring the sources to build without kernel modules, the path
to the kernel sources was not set and the kernel header files were still
checked.

Now, we do not check the kernel sources for compatibility if we're not
going to build any kernel modules.

Signed-off-by: Oliver Smith <oliver@8.c.9.b.0.7.4.0.1.0.0.2.ip6.arpa>
Signed-off-by: Jozsef Kadlecsik <kadlec@blackhole.kfki.hu>
---
 configure.ac | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/configure.ac b/configure.ac
index 0eb1022..d56328c 100644
--- a/configure.ac
+++ b/configure.ac
@@ -167,6 +167,8 @@ AC_CHECK_TYPES([union nf_inet_addr],,,[#include <linux/types.h>
 dnl Checks for functions
 AC_CHECK_FUNCS(gethostbyname2)
 
+if test "$BUILDKMOD" == "yes"
+then
 dnl Check kernel incompatibilities... Ugly like hell
 AC_MSG_CHECKING([kernel source for struct xt_action_param])
 if test -f $ksourcedir/include/linux/netfilter/x_tables.h && \
@@ -309,6 +311,7 @@ else
 	AC_MSG_RESULT(no)
 	AC_MSG_ERROR([Netns support is required in the Linux kernel tree])
 fi
+fi
 
 dnl Checks for compiler characteristics.
 dnl Check extra warning flags except
-- 
1.8.4

